### PR TITLE
Add EGL replay implementation

### PR DESCRIPTION
**Pull request by galpeter**

### CODE REVIEW

File: renderdoc/driver/gl/gl_replay_egl.cpp [hunk 1]
```diff
@@ -94,6 +222,159 @@ ReplayCreateStatus GL_CreateReplayDevice(const char *logfile, IReplayDriver **dr
...
+    {
+      RDCERR(
+          "Couldn't find required function addresses, eglGetProcAddress eglCreateContext"
+          "eglSwapBuffers (etc.)");
+      return eReplayCreate_APIInitFailed;
+    }
+  }
+
+  GLInitParams initParams;
+  RDCDriver driverType = RDC_OpenGL;
```
Comment (baldurk) [inline renderdoc/driver/gl/gl_replay_egl.cpp]: This raises a question - do we want to have captures be cross-compatible between GLES and Desktop GL? i.e. being able to capture on one and replay on another.

I think it'd be a useful feature, I'm just not sure if it's practical. You'd be able to say better than me whether concepts can be translated back and forth, but I'd guess it's common to use extensions that won't have any equivalent on desktop.

If they're going to be distinct (ie. GLES can only be replayed on GLES devices) then you should add a new driver type and set it here, with the name set to OpenGL ES.

File: renderdoccmd/renderdoccmd_linux.cpp [hunk 2]
```diff
@@ -708,6 +708,23 @@ extern "C" void glXWaitX();
...
 #endif
 
+#if defined(RENDERDOC_SUPPORT_GL)
+
+// symbol defined in libGL but not librenderdoc.
+// Forces link of libGL after renderdoc (otherwise all symbols would
+// be resolved and libGL wouldn't link, meaning dlsym(RTLD_NEXT) would fai
+extern "C" void glXWaitX();
+
+#endif
```
Comment (baldurk) [inline renderdoccmd/renderdoccmd_linux.cpp]: This #if segment is identical to the one above?

File: renderdoc/driver/gl/gl_replay_egl.cpp (hunk 1 above)
Comment (galpeter) [inline renderdoc/driver/gl/gl_replay_egl.cpp]: Yes, I'm planning adding a new driver type (in fact I'm basically done with it). However I did not wanted to add that "extra code" into this PR.

File: renderdoccmd/renderdoccmd_linux.cpp (hunk 2 above)
Comment (galpeter) [inline renderdoccmd/renderdoccmd_linux.cpp]: hmmm... strange will fix this asap.

**Comment by galpeter** [comment]
I've updated the PR.

For the dependency issue that is a real concern: the new header dependencies in this PR are what comes from the `official/eglplatform.h`, there are `X11` headers and others. Currently the eglplatform.h is only used for the `EGL_CAST` macro. It would be great if we could detect if the system has EGL libs and headers we could use the cmake's detection abilities however the egl does not have the correct cmake files in place. We would need to add our own implementation to detect the required libs.

**Comment by baldurk** [comment]
Thanks.

I'm not too worried about the dependencies right now - if someone opts into building EGL then we can assume that it's they will figure out what's wrong if there's an error. I'm more concerned with the dependencies required out-of-the-box, since this affects anyone packaging it for distribution, or for the builds that I make. If/when we enable it for linux then we can worry about it - but I think it's most important and most useful for android where the dependencies will definitely be there.

**Pull request merged**